pyuvdata/uvdata/mir_meta_data.py

```diff
@@ -0,0 +1,3087 @@
+            in this object, an error is raised. Default is False, which means that
+            if no match is found for a particular entry, the method quitely moves on
+            to the next item.
+
+        Raises
+        ------
+        ValueError
+            If the keys of `update_dict` are not str or tuples of str, or if no matching
+            field names are found and `raise_err` is set to True.
+        """
+        rebuild_index = False
+        for field, data_dict in update_dict.items():
+            if not isinstance(field, (str, tuple)):
+                raise ValueError(
+                    "update_dict must have keys that are type str or tuples of str."
+                )
+
+            # Check if we have a tuple, since it changes the logic a bit
+            is_tuple = isinstance(field, tuple)
+            # Check if we have a match for the field name
+            has_match = True
+            for item in field if is_tuple else [field]:
+                has_match &= item in self.dtype.fields
+
+            # If no match, and we want to raise an error, do so now.
+            if not has_match:
+                if raise_err:
+                    raise ValueError("Field group %s not found in this object." % field)
+                # Otherwise, just move along.
+                continue
+
+            # Check if we are modifying an index field, which will force us to reindex
+            rebuild_index |= np.any(np.isin(field, self._header_key))
+            rebuild_index |= np.any(np.isin(field, self._pseudo_header_key))
+
+            # Get the existing metadata now, returned as tuples to make it easy
+            # to use the update_dict
+            iter_data = self.get_value(field, return_tuples=is_tuple, use_mask=False)
+
+            # Note that with a complex dtype, passing _data a str will return a
+            # reference to the array we want, so we can update in situ.
+            arr_data = (
+                [self._data[item] for item in field] if is_tuple else self._data[field]
+            )
+
+            # Now go through each value (or tuple of values) and plug in updates.
+            for idx, old_vals in enumerate(iter_data):
+                try:
+                    if not is_tuple:
+                        arr_data[idx] = data_dict[old_vals]
+                    else:
+                        for subarr, new_val in zip(arr_data, data_dict[old_vals]):
+                            subarr[idx] = new_val
+                except KeyError:
+                    # If no matching key, then there is no update to perform
+                    continue
+
+        # If we have messed with an indexing field, then rebuild the header key index.
+        if rebuild_index:
+            self._set_header_key_index_dict()
+
+    def _add_check(self, other, merge=None, overwrite=None, discard_flagged=False):
+        """
+        Check if two MirMetaData objects conflicting header key values.
```

Comment:
missing word. Maybe "...objects contain conflicting..."

Reply:
Fixed in a2f46b4b.

Reply:
Fixed in a2f46b4b.